packet/baseapi.py

```diff
@@ -38,66 +38,91 @@ def call_api(self, method, type="GET", params=None):  # noqa
         if params is None:
             params = {}
 
-        url = "https://" + self.end_point + "/" + method
-
-        headers = {
-            "X-Auth-Token": self.auth_token,
-            "X-Consumer-Token": self.consumer_token,
-            "Content-Type": "application/json",
-        }
+        paginated_data = {}
+        page = 0
+
+        while True:
+            if page == 0 or (paginated_data is not None and paginated_data["meta"]["next"] is None):
+                url = "https://" + self.end_point + "/" + method
+            else:
+                # Appended that last bit b/c w/o it requests weren't coming in correctly
+                url = "https://" + self.end_point + paginated_data["meta"]["next"]["href"] + f"&per_page={params['per_page']}"
```

Comment:
this can all be trimmed down to using the page number instead of the page href, and start at page = 1 to avoid dealing with a separate initial case.